jekyll/_cci2/concepts.md

```diff
@@ -0,0 +1,209 @@
+---
+layout: classic-docs
+title: "Concepts"
+short-title: "Concepts"
+description: "CircleCI 2.0 concepts"
+categories: [getting-started]
+order: 1
+---
+
+*[Basics]({{ site.baseurl }}/2.0/concepts/) > Concepts*
+
+This document provides an overview of the concepts used in CircleCI 2.0 in the following sections:
+
+* TOC
+{:toc}
+
+## Steps
+
+Steps are actions that need to be taken to perform your build. ![step illustration]( {{ site.baseurl }}/assets/img/docs/concepts_step.png) Steps are usually a collection of executable commands. For example, the `checkout` step checks out the source code for a job over SSH. Then, the `run` step executes the `make test` command using a non-login shell by default.
+
+
+```YAML
+...
+    steps:
+      - checkout # Special step to checkout your source code
+      - run: # Run step to execute commands, see
+      # circleci.com/docs/2.0/configuration-reference/#run
+          name: Running tests
+          command: make test # executable command run in
+          # non-login shell with /bin/bash -eo pipefail option
+          # by default.
+...          
+```          
+
+## Image
+
+An image is a packaged system that has the instructions for creating a running container. ![image illustration]( {{ site.baseurl }}/assets/img/docs/concepts_image.png)
+ The Primary Container is defined by the first image listed in `.circleci/config.yml` file. This is where commands are executed for jobs using the Docker executor.
+
+ ```YAML
+ version 2
+ jobs:
+   build1: # job name
+     docker: # Specifies the primary container image,
+     # see circleci.com/docs/2.0/circleci-images/ for
+     # the list of pre-built CircleCI images on dockerhub.
+       - image: buildpack-deps:trusty
+
+       - image: postgres:9.4.1 # Specifies the database image
+        # for the secondary or service container run in a common
+        # network where ports exposed on the primary container are
+        # available on localhost.
+         environment: # Specifies the POSTGRES_USER authentication
+          # environment variable, see circleci.com/docs/2.0/env-vars/
+          # for instructions about using environment variables.
+           POSTGRES_USER: root
+...
+   build2:
+     machine: # Specifies a machine image that uses
+     # an Ubuntu version 14.04 image with Docker 17.06.1-ce
+     # and docker-compose 1.14.0, follow CircleCI Discuss Announcements
+     # for new image releases.
+       image: circleci/classic:201708-01
+...       
+   build3:
+     macos: # Specifies a macOS virtual machine with Xcode version 9.0
+       xcode: "9.0"       
+ ...          
+ ```
+
+
+## Jobs
+
+Jobs are a collection of steps. The Job Space is all of the containers being run by an executor (`docker`, `machine`, or `macos`) for the current job.
```

Comment:
` The Job Space is all of the containers being run by an executor (`docker`, `machine`, or `macos`) for the current job.` -> Not sure about this.

Reply:
I removed it for now and replaced it with "Jobs are a collection of steps and each job must declare an executor that is either `docker`, `machine`, or `macos`. Machine includes a default image if not specified, for Docker and macOS, you must also declare an image."